src/services/ArticleService.js

```diff
@@ -4,11 +4,23 @@ import { backendAPI } from "../config.json";
 export class ArticleService {
 	BASE_URL = "article/v1/articles";
 
+	static buildArticleURL(article) {
+		return `${encodeURIComponent(article.title.replace(/\s/g, "-").toLowerCase())}-${article.id}`;
```

Comment:
Personally, I would like to see the `article.id` run through a hashing function to get a non-sequential value in the url, with the only reason being to eliminate any sort of perception of importance which sequential id's can often infer.

I would also strip out any non-alphanumeric symbols ( :, ;, ', `, etc) that may be present in titles to make them both visually pleasing and url safe.